packages/block-editor/src/components/block-unsaved-changes-indicator/parent-block-selector-unsaved-changes-indicator.js

```diff
@@ -0,0 +1,46 @@
+/**
+ * WordPress dependencies
+ */
+import { Fill } from '@wordpress/components';
+import { useEffect } from '@wordpress/element';
+
+function UnsavedChangesIndicator() {
+	useEffect( () => {
+		// add class to the parent selector to increase it's width
+		const parentSelectorBtn = document.querySelector(
+			'.block-editor-block-parent-selector'
+		);
+		parentSelectorBtn.classList.add( 'parent-block-has-changes' );
+		// add class to the contextual toolbar to move it to the right side
+		const contextualToolBar = document.querySelector(
+			'.block-editor-block-contextual-toolbar'
+		);
+		contextualToolBar.classList.add( 'parent-block-has-changes' );
+
+		return () => {
+			// remove classes from the parent selector and contextual toolbar when component unmounts
+			document
+				.querySelector( '.block-editor-block-parent-selector' )
+				.classList.remove( 'parent-block-has-changes' );
+			document
+				.querySelector( '.block-editor-block-contextual-toolbar' )
+				.classList.remove( 'parent-block-has-changes' );
+		};
+	}, [] );
```

Comment:
I'm not in favour of the modification of other parent elements when this renders. Is there a way to achieve the same result without that being required?

Reply:
> Since the parent selector button is positioned absolutely to the contextual toolbar, it leads to a state like this

Interesting that this is positioned absolutely. That sounds like the core of the problem. Is there a way to style it so that it's an inline element?